File: LV1/1.6/rijeci/rijeci.rb
def gen_fibo
  fib = [1] * 50
  fib[0] = 0

  (2..50).each do |i|
    fib[i] = fib[i - 1] + fib[i - 2]
  end

  fib
end

fib = gen_fibo
n = gets.to_i

puts "#{fib[n - 1]} #{fib[n]}"

File: LV1/1.6/rijeci/rijeci.rs
use std::io;

fn gen_fibo() -> Vec<i64> {
    let mut fib = vec![0, 1];

    for i in 2..50 {
        let cur: i64 = fib[i - 1] + fib[i - 2];
        fib.push(cur);
    }

    return fib;
}

fn main() {
    let fib = gen_fibo();

    let stdin = io::stdin();
    let mut buff = String::new();
    stdin.read_line(&mut buff).unwrap();

    let n: usize = buff.trim().parse().unwrap();

    println!("{} {}", fib[n - 1], fib[n]);
}
